Add simple DQL example to show how to insert and query via DQL

nicholastmosher opened the pull request:
Closes #1, showing how to use the Ditto SDK with DQL to insert and query documents!

- Add new bin to repo, `simple-dql`
- Include a CLI with help text and two subcommands, `create-car` and `query-cars`
- Update the README to show how to execute specific bins with args

Commit:
Rename main bin to simple-querybuilder to make room for more bins

Commit:
Add simple DQL example with insert and select

danielhenrymantilla commented on src/bin/simple_dql.rs:
```diff
@@ -0,0 +1,159 @@
+///
+/// Example: Insert a new car to the "cars" collection
+///
+/// > simple_dql create-car --make=ford --year=2016 --color=blue
```
```suggestion
/// > `simple_dql create-car --make=ford --year=2016 --color=blue`
```

danielhenrymantilla commented on src/bin/simple_dql.rs:
```diff
@@ -0,0 +1,159 @@
+///
+/// Example: Query blue cars from the "cars" collection
+///
+/// > simple_dql query-cars --color=blue
```
```suggestion
/// > `simple_dql query-cars --color=blue`
```

danielhenrymantilla commented on src/bin/simple_dql.rs:
```diff
@@ -0,0 +1,159 @@
+/// If you have not set up ENV variables with your AppID and Token,
+/// you can alternatively pass them as arguments like this:
+///
+/// > simple_dql --app-id="YOUR_APP_ID" --token="YOUR_PLAYGROUND_TOKEN" create-car --make=ford --year=2016 --color=blue
```
```suggestion
/// > `simple_dql --app-id="YOUR_APP_ID" --token="YOUR_PLAYGROUND_TOKEN" create-car --make=ford --year=2016 --color=blue`
```

danielhenrymantilla commented on src/bin/simple_dql.rs:
```diff
@@ -0,0 +1,159 @@
+/// Args needed for any command,
+#[derive(Debug, Parser)]
+struct Args {
+    /// The Ditto App ID to sync with (found at portal.ditto.live)
```
```suggestion
    /// The Ditto App ID to sync with (found at <https://portal.ditto.live>)
```

danielhenrymantilla commented on src/bin/simple_dql.rs:
```diff
@@ -0,0 +1,159 @@
+    #[clap(long, env = "APP_ID")]
+    app_id: String,
+
+    /// The Playground token used to authenticate (found at portal.ditto.live)
```
```suggestion
    /// The Playground token used to authenticate (found at <https://portal.ditto.live>)
```

danielhenrymantilla commented on src/bin/simple_dql.rs:
```diff
@@ -0,0 +1,159 @@
+}
+
+async fn query_cars(store: &Store, color: &str) -> Result<QueryResult> {
+    let query_args = QueryArguments::from(serde_json::json!({
+        "color": color,
+    }));
```
Tiny æsthetical preference for moving the `.into()` to the callsite itself:

```suggestion
    let query_args = serde_json::json!({
        "color": color,
    });
```

danielhenrymantilla commented on src/bin/simple_dql.rs:
```diff
@@ -0,0 +1,159 @@
+
+    // Execute a DQL query and get a result set
+    let result_set = store
+        .execute("SELECT * FROM cars where color = :color", Some(query_args))
```
```suggestion
        .execute("SELECT * FROM cars where color = :color", Some(query_args.into()))
```

danielhenrymantilla reviewed (commented):
Great! ✅

danielhenrymantilla reviewed (approved):
Great! ✅

Commit:
Apply suggestions from code review

- Switch `QueryArguments::from` to `args.into()`
- Add better link formatting for portal

Co-authored-by: Daniel Henry-Mantilla <daniel.henry.mantilla@gmail.com>

Commit:
Add backticks around sample commands

Co-authored-by: Daniel Henry-Mantilla <daniel.henry.mantilla@gmail.com>
